state/accountsDB.go

```diff
@@ -1283,12 +1292,14 @@ func (adb *AccountsDB) snapshotUserAccountDataTrie(
 
 		stats.NewSnapshotStarted()
 
+		iteratorChannelsCopy := iteratorChannels
+		iteratorChannelsCopy.LeavesChan = nil
```

Comment:
Maybe create a new TrieIteratorChannels struct here, and assign the existing err chan to it instead.

Reply:
this solved the other issue also, thanks